### PR TITLE
Docs(DG): Update storage diagram and add alias diagram/manual testing

### LINKED ISSUE #349: Manual testing instruction: Linh

### LINKED ISSUE #357: DG: Update Storage component

### PR DESCRIPTION
**Pull request by cnlinh**
## What this does
Closes #357 #349 #314 and #192
## How to test
1. cd to `docs/` folder of the project.
2. Run `bundle exec jekyll serve`.
3. Open the jekyll-hosted website in your browser (default address: http://127.0.0.1:4000).
4. Check that the DG has the changes

### CODE REVIEW

File: docs/diagrams/alias/AliasMappingClassDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    command : String
+    {static} isValidName(String) : boolean
+    {static} isValidCommand(String) : boolean
+    getAliasName() : String
+    getCommand() : String
+}
+show Alias members
+
+AddressBook *-right-> AliasMapping
+AliasMapping o-right-> "*" Alias
```
Comment (DrWala) [inline docs/diagrams/alias/AliasMappingClassDiagram.puml:34]: Similarly, would suggest changing this to composition.

In addition, perhaps it should be a dotted arrow as the main mode of interaction is through the hashmap, and so, similar to the storage diagrams, it should be a dotted arrow.

Ref: https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/uml.html#what-4

**Review by DrWala — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -376,29 +376,32 @@ The diagram below details how the user's command to add an issue propagates thro
 The `Alias` feature allows users to define a shortcut for a longer command that is often used. The longer command can then be executed by entering the alias instead of the full or partial command.
 
 #### Implementation
-User-defined `Alias` is stored in `AliasMapping` within `UserPrefs`. `AliasMapping` internally uses `HashMap<String, Alias>` to store the mapping between the name of an `Alias` object and itself. With `AliasMapping` included in `UserPrefs`, `UserPrefs` supports the following methods:
+User-defined `Alias` is stored in `AliasMapping` within `AddressBook`. `AliasMapping` internally uses `HashMap<String, Alias>` to store the mapping between the name of an `Alias` object and itself. The class diagram of `AliasMapping` and `Alias` is shown below:
```
[the remark this answers is not in the document]
Comment (cnlinh) [reply docs/DeveloperGuide.md:379]: Updated!

File: docs/diagrams/StorageClassDiagram.puml [hunk 3]
```diff
@@ -7,18 +7,28 @@ skinparam classBackgroundColor STORAGE_COLOR
...
+Interface CommandHistoryStorage <<Interface>>
 
 Class StorageManager
-Class JsonUserPrefsStorage
-Class JsonAddressBookStorage
-
 StorageManager .left.|> Storage
 StorageManager o-right-> UserPrefsStorage
 StorageManager o--> AddressBookStorage
+StorageManager o-up-> CommandHistoryStorage
```
[the remark this answers is not in the document]
Comment (cnlinh) [reply docs/diagrams/StorageClassDiagram.puml:16]: Updated!

File: docs/diagrams/alias/AliasMappingClassDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+@startuml
+!include ../style.puml
+
+skinparam arrowThickness 1.1
```
Comment (DrWala) [inline docs/diagrams/alias/AliasMappingClassDiagram.puml:4]: Could you follow the general class diagram styling?

```
skinparam arrowThickness 1.1
skinparam arrowColor MODEL_COLOR
skinparam classBackgroundColor MODEL_COLOR
```

**Review by DrWala — CHANGES_REQUESTED** [review]

**Review by DrWala — APPROVED** [review]

**Pull request merged**